iothub/device/src/AmqpConnectionPoolSettings.cs

```diff
@@ -1,81 +1,96 @@
 
+using System;
+
 namespace Microsoft.Azure.Devices.Client
 {
-    using System;
-
     /// <summary>
     /// contains Amqp Connection Pool settings for DeviceClient
     /// </summary>
-
     public sealed class AmqpConnectionPoolSettings
     {
-        static readonly TimeSpan DefaultConnectionIdleTimeout = TimeSpan.FromMinutes(2);
-        static readonly TimeSpan MinConnectionIdleTimeout = TimeSpan.FromSeconds(5);
-        const uint DefaultPoolSize = 100; // This will allow upto 100000 devices
-        const uint MaxNumberOfPools = ushort.MaxValue;
+        private static readonly TimeSpan s_defaultConnectionIdleTimeout = TimeSpan.FromMinutes(2);
+        private static readonly TimeSpan s_minConnectionIdleTimeout = TimeSpan.FromSeconds(5);
+        private uint _maxPoolSize;
+        private TimeSpan _connectionIdleTimeout;
         internal const uint MaxDevicesPerConnection = 995; // IotHub allows upto 999 tokens per connection. Setting the threshold just below that.
 
-        uint maxPoolSize;
-        TimeSpan connectionIdleTimeout;
+        /// <summary>
+        /// The default size of the pool
+        /// </summary>
+        /// <remarks>
+        /// Allows up to 100,000 devices
+        /// </remarks>
+        private const uint DefaultPoolSize = 100;
+
+        /// <summary>
+        /// The maximum value that can be used for the MaxPoolSize property
+        /// </summary>
+        public const uint AbsoluteMaxPoolSize = ushort.MaxValue;
 
+        /// <summary>
+        /// Creates an instance of AmqpConnecitonPoolSettings with default properties
+        /// </summary>
         public AmqpConnectionPoolSettings()
         {
-            this.maxPoolSize = DefaultPoolSize;
-            this.Pooling = false;
-            this.connectionIdleTimeout = DefaultConnectionIdleTimeout;
+            _maxPoolSize = DefaultPoolSize;
+            Pooling = false;
+            _connectionIdleTimeout = s_defaultConnectionIdleTimeout;
         }
 
+        /// <summary>
+        /// The maximum pool size
+        /// </summary>
         public uint MaxPoolSize
         {
-            get { return this.maxPoolSize; }
+            get => _maxPoolSize;
 
             set
             {
-                if (value > 0 && value <= MaxNumberOfPools)
-                {
-                    this.maxPoolSize = value;
-                }
-                else
-                {
-                    throw new ArgumentOutOfRangeException(nameof(value));
-                }
+                _maxPoolSize = value > 0 && value <= AbsoluteMaxPoolSize
+                    ? value
+                    : throw new ArgumentOutOfRangeException(nameof(value));
             }
         }
 
+        /// <summary>
+        /// Whether or not to use connection pooling
+        /// </summary>
         public bool Pooling { get; set; }
 
+        /// <summary>
+        ///
```

Comment:
empty summary?

Reply:
I probably paused to inspect what it really was, and didn't get back to it. What is it, exactly?

Reply:
The name would suggest that it is being used to validate client side connection heartbeat interval, and gracefully close the connection if no frames are sent for the threshold specified. However, @davilu recently added another property called 'IdleTimeout`, which seems to be targeting similar behavior.
@davilu - is there a different use case for `ConnectionIdleTimeout`? I cannot find it referenced anywhere explicitly, other than in the `Equals` method below.

Reply:
I don't see the property being referenced outside the class except by a test method. Am I missing something?